### DISCUSSION TITLE
Feature Request: Add section folding to the editor

### DISCUSSION THREAD

**Discussion by davistech** [notaapp/talk#219] [Feature Requests] [answered]
It would help to have the ability to fold at the header level. Even better, at the bullet level. An example can be found in Obsidian and Marked 2.

Thank you.

**Comment by led** [comment]
Also: #18

**Comment by erusev** [comment]
Hi @davistech, and thanks for the feature request!

We've been researching this feature for a while, and we'd love to get more input on it.

What are your use cases for folding (the scenarios where folding is helpful)?

Thanks!

**Comment by davistech** [comment]
Thanks for the quick response. I’ve just added Nota to my collection of markdown apps and am very impressed with it, but need the folding to be able to use it as my primary app.

I teach free workshops on various Mac topics to local groups (mostly retirees) and the technique I’ve developed for these is as follows. I use Keynote for formal presentations, but now use a combination of Obsidian + Multimarkdown Composer + Marked 2 for the workshops (taught on Zoom).
Obsidian has folding which is helpful when creating a workshop — helps focus on specific topics.
Multimarkdown Composer is my current default markdown external editor from Obsidian.
Marked 2 (Brett Terpstra) renders the markdown as RTF and all sections are foldable.
I can also present from Obsidian Preview mode and the sections are foldable which makes it easy to focus on the topics while presenting.

If Nota had folding then I would be able to use it as my primary markdown editor and bypass steps 2 and 3 in the workflow.

Obsidian works fine for presenting, but requires tweaking the CSS to render the font larger.

Let me know if I can provide more information.

Phil

**Comment by erusev** [comment]
Phil, thanks for sharing this!

> Obsidian has folding which is helpful when creating a workshop — helps focus on specific topics.

Have you experimented with alternative ways (that don't involve folding) to achieve the same (focus on a specific topic)? For example, breaking the document into files, where each topic is a separate file (that you can easily focus on).

**Comment by davistech** [comment]
> On Jun 24, 2021, at 1:11 PM, Emanuil Rusev ***@***.***> wrote:
> 
> 
> Phil, thanks for sharing this!
> 
> Obsidian has folding which is helpful when creating a workshop — helps focus on specific topics.
> 
> Have you experimented with alternative ways (that don't involve folding) to achieve the same (focus on a specific topic)? For example, breaking the document into files, where each topic is a separate file (that you can easily focus on).
> 
> 

Yes. I’ve done that for years, but my current workflow is much more efficient and makes the process simpler. Easier to create, easier to update, easier to present, and easier to create notes and handouts for students.

Phil

**Comment by erusev** [comment]
Phil, thanks for elaborating on that. This might be hard to answer, but do you have an explanation for what makes folding more efficient (or simpler)? One benefit I know about folding is that it makes it possible to see the 1st item and the 3rd item next to each other (because you collapsed the 2nd item). I wouldn't know how to achieve this using files. I'm looking for things like tihs.

**Comment by davistech** [comment]
I collapse all sections, then open the section that I’m editing or presenting. I can collapse at the header level or at the list level (when there are sublists).

If you look at how Obsidian or Marked works you can see examples. I can do some screen captures to illustrate but it won’t be for a couple of days because I’m prepping for my next workshop tomorrow. Also I don’t want the images posted anywhere — for your use only.

Phil

**Comment by erusev** [comment]
Phil, thanks for this!

> I can do some screen captures to illustrate but it won’t be for a couple of days because I’m prepping for my next workshop tomorrow.

That would great! You can send it to support@nota.md -- we can delete it once we've had a look.

**Reply by HusseinT2000** [reply]
Hey, just wanted to say that I'm hoping this eventually gets implemented as it helps break down topics.

Also allows the ability to test yourself as the parent can act as a front of the flashcard and the daughter as the answer?

**Reply by astoilkov** [reply]
We are currently working on it. It's possible we introduce folding of lists in the next release (as an experiment in Preferences → Labs).

**Comment by davistech** [comment]
Eagerly looking forward to this feature!

**Comment by davistech** [comment, marked as answer]
**The folding in ver 37 works great!** Thanks for adding this feature.

**Reply by astoilkov** [reply]
Happy to hear that 😊

**Comment by Arbiter313** [comment]
This feature is fantastic but is there a particular reason why it is not possible to fold H1 headings too? That would also be useful

**Reply by astoilkov** [reply]
It's just an experiment. When we were using it we were doing ⌘A (Select All) and ⌘' (Fold) to fold everything. However, folding the H1 wasn't very useful for us as it was hiding the entire document.

We recognize that this isn't consistent with other apps and are open to changing it.

**Reply by Arbiter313** [reply]
I see. I think it comes down to a philosophical difference about how H1 headings are used in different use cases! Perhaps a way to address this is for the fold all shortcut to apply to H1 headings but also offer shortcuts that fold at the H2 and H3 heading levels.

**Reply by astoilkov** [reply]
Agree. We will try to improve on this in the next release. Thanks!

**Reply by lewallen** [reply]
Nota is a beautiful app, I would definitely purchase a license if it could fold H1 headings (for those of us who dont use it as a unique global heading), and also if there weren't a lower limit to the font size (though the folding issue is the deal breaker). Any chance you could implement those? Thanks for your time!
